spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -85,67 +257,181 @@ def write_recording(recording, save_path, acquisition_name='ElectricalSeries', *
+                          'session_start_time': datetime.now()}
+                kwargs.update(**nwbfile_kwargs)
+                nwbfile = NWBFile(**kwargs)
 
-        rate = recording.get_sampling_frequency()
-        ephys_data = recording.get_traces().T
-
-        ephys_ts = ElectricalSeries(
-            name=acquisition_name,
-            data=ephys_data,
-            electrodes=electrode_table_region,
-            starting_time=recording.frame_to_time(0),
-            rate=rate,
-            resolution=1e-6,
-            comments='Generated from SpikeInterface::NwbRecordingExtractor',
-            description='acquisition_description'
-        )
+            # Tests if Device already exists
+            aux = [isinstance(i, Device) for i in nwbfile.children]
+            if any(aux):
+                device = nwbfile.children[np.where(aux)[0][0]]
+            else:
+                device = nwbfile.create_device(name='Device')
+
+            # Check if 'groups' property exists for channels in Recording
+            if 'group' in recording.get_shared_channel_property_names():
+                el_groups_names = np.unique(recording.get_channel_groups()).tolist()
+            else:
+                el_groups_names = ["0"]
+                # Electrode groups are required for NWB, for consistency we create group for Recording channels
+                ids = recording.get_channel_ids()
+                vals = [0] * len(ids)
+                recording.set_channel_groups(channel_ids=ids, groups=vals)
+            # Creates electrode groups for groups names in Recording
+            for grp_name in el_groups_names:
+                if str(grp_name) not in nwbfile.electrode_groups:
+                    elec_group = nwbfile.create_electrode_group(
+                        name=str(grp_name),
+                        location="electrode_group_location",
+                        device=device,
+                        description="electrode_group_description"
+                    )
+
+            # Check for existing electrodes
+            if nwbfile.electrodes is not None:
+                nwb_elec_ids = nwbfile.electrodes.id.data[:]
+            else:
+                nwb_elec_ids = []
+
+            # add new electrodes with id, (x, y, z) and groups
+            for m in channel_ids:
+                if m not in nwb_elec_ids:
+                    location = recording.get_channel_property(m, 'location')
+                    while len(location) < 3:
+                        location = np.append(location, [0])
+                    impedence = -1.0
+                    grp_name = recording.get_channel_groups(channel_ids=[m])
+                    grp = nwbfile.electrode_groups[str(grp_name[0])]
+                    nwbfile.add_electrode(
+                        id=m,
+                        x=float(location[0]), y=float(location[1]), z=float(location[2]),
+                        imp=impedence,
+                        location='unknown',
```

Comment:
I think it would be better, in case the location is unkown, not to save it asa property. This might confuse some sorters that only check if the 'location' property is present

Reply:
As suggested in the next comment, I think it's better to load the locations as 2D rather than 3D

Reply:
@luiztauffer @alejoe91 Currently there is an issue with location in NWB that it is actually meant to be the electrode position in the space of the brain, not on the electrodes. This is a problem because the location in the brain is often not known, while the position on the electrode is needed for spike sorting. I opened this issue (https://github.com/NeurodataWithoutBorders/nwb-schema/issues/325)and linked PR last week. This would use rel_x, rel_y, and rel_z columns in the electrodes table to mark the position of the electrode on the `ElectrodeGroup`. These columns would each be optional, so you could just define rel_x and rel_y, and leave out rel_z. This solves our problem of confusion between brain space and probe space, and it would solve your problem of not wanting to deal with a third dimension.

Reply:
I see. So could we use rel_x, rel_y and rel_z already?